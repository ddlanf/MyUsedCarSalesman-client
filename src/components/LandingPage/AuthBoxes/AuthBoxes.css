.auth-boxes > *{
  box-sizing: border-box;
}

.auth-boxes > div{
  height: 35vw;
  background-color: #EAEDEC;
  border: #999090 solid 2px;
}
  
.auth-boxes{
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  height: 45vw;
  width: 100%;
}
  
.Login-register-admin{
  font-size: 2vw;
  text-align: left;
  width: 50%;
}

.additional-info{
  width: 25%;
}
  
.Login-register-admin > p{
 font-size: 2vw;
 text-align: center;
 line-height: normal;
 margin: auto;
}

.mini-login-header{
 text-align: center;
 margin-top: 2%;
 margin-bottom: 0;
}

.mini-login-label, .mini-login-input{
 display: block;
 margin-right: auto;
 margin-left: auto;
 width: 60%;
 margin-bottom: 1%;
 background-color: 999090;
}
  
.mini-login-label{
  font-size: 1.7vw;
}

.mini-login-input{
  font-size: 1.7vw;
}

.mini-login-submit{
  height: 4vw;
  width: 20%;
  text-align: center;
  font-size: 1.4vw;
  padding: 0;
  margin-top: 3%;
  margin-bottom: 3%;
  margin-left: auto;
  margin-right: auto;
  display: block;
  border-radius: 10%;
  border: #999090 solid 1px;
  background-color: white;
}

.mini-login-submit:hover{
  background-color: #999090;
  transition: 0.3s;
}

.mini-login-submit:active{
  border-top: grey solid 2px;
  background-color: #999090;
  padding-top: 0.5%;
}

.mini-login-register-heading{
  font-size: 5vw;
  width: 90%; 
  margin-left: auto;
  margin-right: auto;
}

.mini-login-link{
  display: block;
  color: blue;
}

.mini-login-link:hover{
  text-decoration: underline;
}

.additional-info{
    padding: 2%;
}
  
.additional-info > p{
     font-size: 1.7vw;
     text-align: center;
}

.auth-box-break{
  display: block;
  margin-top: 1%;
  padding: 0;
}
.mini-login-admin{
  padding-top: 1vw;
}

.github, .linkedIn{
  width: 30%;
  margin: 10%;
}

.image-link-container{
  text-align: center;
  width: 100%;
  display: block;
}

.portfolio-link{
  font-weight: bold;
  color: blue;
  margin-top: 1rem;
  margin-bottom: 0;
}

.portfolio-link:hover{
  text-decoration: underline;
}

/*for demo purposes only*/
.demo > h1{
  font-size: 3vw;
  margin: 0;
}

.demo-break{
  display: block;
}

.demo{
  width: 80%;
  border: solid black 1px;
  top: 4%;
  right: 2%;
  background-color:  #EAEDEC;
  text-align: center;
  margin-left: auto;
  margin-right: auto;
  font-size: 2.5vw;
  line-height: 4vw;
  margin-bottom: 8%;
}

ol{
  text-align: left;
  margin-left: 10%;
}
