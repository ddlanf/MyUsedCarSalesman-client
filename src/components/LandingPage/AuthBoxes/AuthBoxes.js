import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import './AuthBoxes.css'
export default class AuthBoxes extends Component {
    render() {
        return (
            <>
               <section className="auth-boxes">
                    <div className="Login-register-admin">
                        <h2 className="mini-login-header">Login</h2>
                        <form>
                            <label name="user-name" className="mini-login-label">User Name/Email</label>
                            <input type="text" className="mini-login-input"></input>
                            <label name="password" className="mini-login-label">Password</label>
                            <input type="text" className="mini-login-input"></input>
                            <input type="submit" className="mini-login-submit"></input>
                        </form>
                            <p className="mini-login-register">
                                New to MyUsedCarSalesman?
                                <span className="auth-box-break">
                                    <Link
                                        to='/register'
                                        className="landing-page-link"
                                        >
                                        Register your account
                                    </Link>
                                </span>
                            </p>
                            <p className="mini-login-admin">
                                <Link
                                    to='/admin-login'
                                    className="landing-page-link"
                                    >
                                Admin Login
                                </Link>
                            </p>
                    </div>
                    <div className="additional-info">
                        <p>MyUsedCarSalesman</p>
                        <p>All rights reserved</p>
                        <p>Created on xx/xx/20xx</p>
                        <p>By Daniel Laufenberg</p>
                        <p>Github Link LinkedIn Link</p>
                    </div>
                </section>  
            </>
        )
    }
}
