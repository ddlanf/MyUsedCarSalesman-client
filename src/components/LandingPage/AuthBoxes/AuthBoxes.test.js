import React from 'react';
import ReactDOM from 'react-dom';
import AuthBoxes from './AuthBoxes';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

describe('AuthBoxes component', () => {
    
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                        <AuthBoxes/>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(
                <BrowserRouter>
                    <AuthBoxes/>
                </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});