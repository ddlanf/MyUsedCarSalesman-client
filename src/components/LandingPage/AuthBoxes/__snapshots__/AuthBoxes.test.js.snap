// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthBoxes component renders the UI as expected 1`] = `
Array [
  <section
    className="auth-boxes"
  >
    <div
      className="Login-register-admin"
    >
      <h2
        className="mini-login-header"
      >
        Login
      </h2>
      <form
        onSubmit={[Function]}
      >
        <label
          className="mini-login-label"
          name="user-name"
        >
          User Name
        </label>
        <input
          className="mini-login-input"
          name="user_name"
          onChange={[Function]}
          type="text"
        />
        <label
          className="mini-login-label"
          name="user-password"
        >
          Password
        </label>
        <input
          className="mini-login-input"
          name="password"
          onChange={[Function]}
          type="password"
        />
        <input
          className="mini-login-submit"
          type="submit"
        />
      </form>
      <p
        className="mini-login-register"
      >
        New to MyUsedCarSalesman?
        <span
          className="auth-box-break"
        >
          <a
            className="mini-login-link"
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
        </span>
      </p>
      <p
        className="mini-login-admin"
      >
        <a
          className="mini-login-link"
          href="/admin-login"
          onClick={[Function]}
        >
          Admin Login
        </a>
      </p>
    </div>
    <div
      className="additional-info"
    >
      <p>
        MyUsedCarSalesman
      </p>
      <p>
        All rights reserved
      </p>
      <p>
        Created on 04/01/2020
      </p>
      <p>
        By Daniel Laufenberg
      </p>
      <p>
        <a
          className="portfolio-link"
          href="https://ddlanf.github.io/Portfolio"
          rel="noopener noreferrer"
          target="_blank"
        >
          Portfolio
        </a>
      </p>
      <div
        className="image-link-container"
      >
        <a
          href="https://github.com/ddlanf"
        >
          <img
            alt="github"
            className="github"
            rel="noopener noreferrer"
            src="github.png"
            target="_blank"
          />
        </a>
        <a
          href="https://www.linkedin.com/in/daniel-laufenberg/"
        >
          <img
            alt="linkedIn"
            className="linkedIn"
            rel="noopener noreferrer"
            src="linkedIn.png"
            target="_blank"
          />
        </a>
      </div>
    </div>
  </section>,
  <div
    className="demo"
  >
    <h1
      onClick={[Function]}
    >
      Click here for instruction
    </h1>
    
  </div>,
]
`;
